### PR TITLE
Feature/update order status by

### PR DESCRIPTION
**Pull request by ney601**

### CODE REVIEW

**Review by Aebel-Shajan — APPROVED** [review]
lgtm

**Review by Jean-005 — APPROVED** [review]
Flawless Yesica!

**Review by sahilpatel1906 — CHANGES_REQUESTED** [review]
Add a optional is empty check

File: capstone_backend/src/main/java/com/example/Capstone_Backend/controllers/OrderController.java [hunk 1]
```diff
@@ -12,5 +17,13 @@ public class OrderController {
     @Autowired
     OrderService orderService;
 
+    @PatchMapping (value = "/{id")
+    public ResponseEntity<Order> updateOrderStatusById(@PathVariable long id, @RequestBody boolean isDelivered){
+        Order updateOrder = orderService.updateOrderStatus(id, isDelivered);
```
Comment (Aebel-Shajan) [inline capstone_backend/src/main/java/com/example/Capstone_Backend/controllers/OrderController.java:22]: Make this optional instead

**Review by Tommybui28 — APPROVED** [review]
ney601 very good

File: capstone_backend/src/main/java/com/example/Capstone_Backend/services/OrderService.java [hunk 2]
```diff
@@ -1,12 +1,23 @@
...
+import org.springframework.web.bind.annotation.PatchMapping;
 
 @Service
 public class OrderService {
     @Autowired
     OrderRepository orderRepository;
 
+
+    public Order updateOrderStatus(long id, boolean isDelivered) {
+        Order orderToUpdate = orderRepository.findById(id).get();
```
Comment (Aebel-Shajan) [inline capstone_backend/src/main/java/com/example/Capstone_Backend/services/OrderService.java:16]: Make this Optional<Order> instead

**Review by Aebel-Shajan — COMMENTED** [review]
Use Optional<Order> in both the order service and the controller.

**Review by sahilpatel1906 — APPROVED** [review]

**Pull request merged**